scripts/import_standard_ebooks.py

```diff
@@ -0,0 +1,158 @@
+import json
+from typing import Dict, List, Optional
+import requests
+import sys
+import time
+
+import os.path as path
+
+import feedparser
+
+from openlibrary.core.imports import Batch
+
+sys.path.insert(0, path.abspath(path.join(path.sep, 'openlibrary')))
+from openlibrary.config import load_config  # noqa: E402
+load_config(
+    path.abspath(path.join(
+        path.sep, 'olsystem', 'etc', 'openlibrary.yml')))
+from infogami import config  # noqa: E402,F401
+
+FEED_URL = 'https://standardebooks.org/opds/all'
+LAST_UPDATED_TIME = './out/standard_ebooks_last_updated.txt'
+
+
+def get_feed():
+    """Fetches and returns Standard Ebook's feed."""
+    r = requests.get(FEED_URL)
+    return feedparser.parse(r.text)
+
+
+def map_data(entry):
+    """Maps Standard Ebooks feed entry to an Open Library import object."""
+    std_ebooks_id = entry.id.replace('https://standardebooks.org/ebooks/', '')
+    mapped_data = {
+        "title": entry.title,
+        "source_records": [f"standardebooks:{std_ebooks_id}"],
+        "publishers": [entry.publisher],
+        "publish_date": entry.dc_issued[0:4],
+        "authors": [{"name": author.name} for author in entry.authors],
+        "description": entry.content[0].value,
+        "subjects": [tag.term for tag in entry.tags],
+        "identifiers": {
+            "standard_ebooks": [std_ebooks_id]
+        }
+    }
+    return mapped_data
+
+
+def create_batch(records: List[Dict[str, str]]) -> None:
+    """Creates Standard Ebook batch import job.
+
+    Attempts to find existing Standard Ebooks import batch.
+    If nothing is found, a new batch is created. All of the
+    given import records are added to the batch job as JSON strings.
+    """
+    now = time.gmtime(time.time())
+    batch_name = f'standardebooks-{now.tm_year}{now.tm_mon}'
+    batch = Batch.find(batch_name) or Batch.new(batch_name)
+    batch.add_items([{
+        'ia_id': r['source_records'][0],
+        'data': json.dumps(r)} for r in records]
+    )
+
+
+def get_last_updated_time() -> Optional[str]:
+    """Gets date of last import job.
+
+    Last updated dates are read from a local file.  If no
+    file exists, None is returned. Last updated date is
+    expected to be in HTTP-date format:
+    https://httpwg.org/specs/rfc7231.html#http.date
+
+    returns last updated date string or None
+    """
+    last_updated = None
+
+    if path.exists(LAST_UPDATED_TIME):
+        with open(LAST_UPDATED_TIME, 'r') as f:
+            last_updated = f.readline()
+
+    return last_updated
+
+
+def find_last_updated():
+    """Fetches and returns Standard Ebooks most recent update date.
+
+    Returns None if the last modified date is not included in the
+    response headers.
+    """
+    r = requests.head(FEED_URL)
+    return r.headers['last-modified'] if r.ok else None
+
+
+def convert_date_string(date_string: str) -> time.struct_time:
+    """date_string will be formatted similarly to this:
+    Fri, 05 Nov 2021 03:50:24 GMT
```

Comment:
Nice spot for a doctest.

Reply:
I tried, but ended up in a no-win situation: The expected value is so long that it causes a linting error.  Using `# noqa` makes the tests fail.